### PR TITLE
Setting RunWorkingDirectory to be the full path using $(TargetDir).

### LINKED ISSUE #301: `RunWorkingDirectory` needs to be a full path
## Steps to reproduce
Fix https://github.com/dotnet/cli/issues/4447
Have a project in a separate directory than the current directory.
`dotnet run3 --project App\App.csproj`

## Expected  behavior
The project should build and run.

## Actual behavior
```
System.ComponentModel.Win32Exception: The directory name is invalid
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Microsoft.DotNet.Cli.Utils.Command.Execute()
   at Microsoft.DotNet.Tools.Run.Run3Command.Start()
   at Microsoft.DotNet.Tools.Run.Run3Command.<>c__DisplayClass0_0.<Run>b__0()
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.DotNet.Tools.Run.Run3Command.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```

The issue here is the RunWorkingDirectory is coming back as a relative path from the .csproj.  We should be returning full path instead.

### PR DESCRIPTION
**Pull request by eerhardt**
Fix #301 

@dsplaisted @nguerrera 

/cc @dotnet/project-system

### CODE REVIEW

**Comment by nguerrera** [comment]
LGTM

**Comment by dsplaisted** [comment]
What about the case the `OutputType` isn't `Exe` (or the `TargetFramework` isn't `.NETFramework` or `.NETCoreApp`?  In that case `RunWorkingDirectory` is set to the value of `StartWorkingDirectory`, which doesn't seem to be defined anywhere.  Do we care what happens there if `StartWorkingDirectory` isn't rooted?

**Comment by eerhardt** [comment]
> What about the case the OutputType isn't Exe (or the TargetFramework isn't .NETFramework or .NETCoreApp?

If that isn't the case, then the project isn't "runnable".

> In that case RunWorkingDirectory is set to the value of StartWorkingDirectory, which doesn't seem to be defined anywhere.

StartWorkingDirectory can be set in the user's .csproj, it doesn't get defaulted by any targets file.

> Do we care what happens there if StartWorkingDirectory isn't rooted?

Good call.  I pushed a new commit to ensure RunWorkingDirectory is always rooted.

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets [hunk 1]
```diff
@@ -140,19 +140,24 @@ Copyright (c) .NET Foundation. All rights reserved.
...
         <RunArguments Condition="'$(RunArguments)' == ''">$(StartArguments)</RunArguments>
-        <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' == ''">$(OutDir)</RunWorkingDirectory>
+        <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' == ''">$(TargetDir)</RunWorkingDirectory>
       </PropertyGroup>
     </When>
   </Choose>
 
+  <PropertyGroup>
+    <!-- Ensure $(RunWorkingDirectory) is a full path -->
+    <RunWorkingDirectory Condition="'$(RunWorkingDirectory)' != '' and '$([System.IO.Path]::IsPathRooted($(RunWorkingDirectory)))' != 'true'">$([System.IO.Path]::GetFullPath('$(MSBuildProjectDirectory)/$(RunWorkingDirectory)'))</RunWorkingDirectory>
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: Is the extra / needed? Isn't MSBuildProjectDirectory guaranteed to have trailing slash? I've found it very good to keep with the convention of having trailing slashes in properties and concatenating them together without slashes. This limits the risk of setting something to near the root of the file system due to an error where a property is empty.

**Pull request merged**